tests/identity/identify/test_protocol.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+
+from libp2p.identity.identify.pb.identify_pb2 import Identify
+from libp2p.identity.identify.protocol import ID, _mk_identify_protobuf
+from libp2p.peer.peerinfo import info_from_p2p_addr
+from tests.utils import set_up_nodes_by_transport_opt
+
+
+@pytest.mark.asyncio
+async def test_identify_protocol():
+    transport_opt_list = [["/ip4/127.0.0.1/tcp/0"], ["/ip4/127.0.0.1/tcp/0"]]
+    (host_a, host_b) = await set_up_nodes_by_transport_opt(transport_opt_list)
+
+    addr = host_a.get_addrs()[0]
+    info = info_from_p2p_addr(addr)
+    await host_b.connect(info)
```

Comment:
I think we can use `host_pair_factory` in `tests/factories.py` here, which makes two connected `BasicHost`s.